app/common/data/interfaces/collections.py

```diff
@@ -314,6 +322,13 @@ def clear_submission_events(submission: Submission, key: SubmissionEventKey, for
 
 
 def add_question_condition(question: Question, user: User, managed_expression: "ManagedExpression") -> Question:
+    if not is_question_dependency_order_valid(question, managed_expression.referenced_question):
+        raise DependencyOrderException(
+            "Cannot add managed condition that depends on a later question",
+            question,
+            managed_expression.referenced_question,
+        )
```

Comment:
Until we've got this in the DB (if we can 👀), really sensible to add this here as a way of making sure (/ as sure as possible) that we don't miss doing this validation somewhere up the chain.